packages/core/src/lib/connection_manager.ts

```diff
@@ -195,7 +195,10 @@ export class ConnectionManager
           conn.tags = Array.from(new Set([...conn.tags, ...tags]));
         });
 
-        this.dialAttemptsForPeer.delete(peerId.toString());
+        // instead of deleting the peer from the peer store, we set the dial attempt to Infinity
+        // this helps us keep track of peers that have been dialed before
+        this.dialAttemptsForPeer.set(peerId.toString(), Infinity);
```

Comment:
To me it seems better to avoid using `Infinity` and instead use `-1` as special value. 
Can't say any hard reasons not to but I rarely see such a thing out there.